src/core/tsi/alts/crypt/aes_gcm.cc

```diff
@@ -35,26 +39,113 @@ constexpr size_t kKdfCounterLen = 6;
 constexpr size_t kKdfCounterOffset = 2;
 constexpr size_t kRekeyAeadKeyLen = kAes128GcmKeyLength;
 
+namespace grpc_core {
+
+GsecKeyFactory::GsecKeyFactory(absl::Span<const uint8_t> key, bool is_rekey)
+    : key_(std::vector<uint8_t>(key.begin(), key.end())), is_rekey_(is_rekey) {}
```

Comment:
micro-nit: Do we need the `std::vector<uint8_t>` here? Is it sufficient to write
```
: key_(key.begin(), key.end())
```
instead?

Reply:
Done. In fact, I also removed `absl::MakeConstSpan` in some (unfortunately not all) places, as the compiler is smart enough to recognize `{key, key_length}` initialization.